src/utils_gen/utils.jl

```diff
@@ -270,3 +270,288 @@ function save_results(model::Model; directory = joinpath(pwd(), "csv_files"))
         YAML.write(io, metadata)
     end
 end
+
+"""
+    get_types(input) -> Vector{Symbol}
+
+Retrieves the names of all defined types from modules or packages.
+
+# Method Overloads
+
+- `get_types(modul::Module)`:
+  Returns a vector of type names defined in the given module.
+
+- `get_types(moduls::Vector{Module})`:
+  Returns a combined vector of type names from multiple modules.
+
+- `get_types(pkg::Union{String, Symbol})`:
+  Converts the package name to a module (via `Main`) and returns its defined types.
+
+- `get_types(pkgs::Union{Vector{<:Union{String, Symbol}}, Set{<:Union{String, Symbol}}})`:
+  Returns a combined vector of type names from multiple packages.
+
+# Arguments
+- `input`: Can be a single module, a vector of modules, a single package name (as `String` or `Symbol`), or a collection of package names.
+
+# Returns
+- `Vector{Symbol}`: A list of names corresponding to types defined in the given module(s) or package(s).
+
+# Description
+This set of functions helps extract type definitions from Julia modules or packages. It filters out non-type bindings and collects only those that are instances of `DataType`.
+
+# Example
+```julia
+get_types(Base)  # returns type names defined in Base
+
+get_types(["Base", "Core"])  # returns type names from both packages
+```
+"""
+function get_types(modul::Module)
+    types = []
+    for name ∈ names(modul)
+        if isdefined(modul, name) && (
+            getfield(modul, name) isa DataType || (
+                getfield(modul, name) isa UnionAll &&
+                Base.unwrap_unionall(getfield(modul, name)) isa DataType
+            )
+        )
+            push!(types, name)
+        end
+    end
+    return types
+end
+
+function get_types(moduls::Vector{Module})
+    types=[]
+    for modul ∈ moduls
+        append!(types, get_types(modul))
+    end
+    return types
```

Comment:
Consider compactification
```suggestion
get_types(modules::Vector{Module}) = [get_types(modul) for modul ∈ modules]
```